8345212: Since checker should better handle non numeric values

nizarbenalla opened the pull request:
Adds a new `-ignoreSince` flag to the SinceChecker, mainly to help long running projects that are not sure about `@since` version of an API.
The value of `@since` in `@since <ignoredString>` will be the release version, as I assume that this is a new API in a different OpenJDK project. This flag is not meant to be used in the JDK.

I also improved error message when the actual `@since` value doesn’t match expectations. It used to say `@since version 9; should be <number>` but it now explains the error more clearly.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345212](https://bugs.openjdk.org/browse/JDK-8345212): Since checker should better handle non numeric values (**Sub-task** - P4)


### Reviewers
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/23447/head:pull/23447` \
`$ git checkout pull/23447`

Update a local copy of the PR: \
`$ git checkout pull/23447` \
`$ git pull https://git.openjdk.org/jdk.git pull/23447/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 23447`

View PR using the GUI difftool: \
`$ git pr show -t 23447`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/23447.diff">https://git.openjdk.org/jdk/pull/23447.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/23447#issuecomment-2636943882)
</details>

liach commented on test/jdk/tools/sincechecker/SinceChecker.java:
```diff
@@ -452,19 +476,38 @@ private void checkElement(Element explicitOwner, Element element, Types types,
     private Version extractSinceVersionFromText(String documentation) {
-        Pattern pattern = Pattern.compile("@since\\s+(\\d+(?:\\.\\d+)?)");
+        for (String ignoreSince : IGNORE_SINCE) {
+            if (documentation.contains(ignoreSince)) {
```
Would it be more efficient to move this equality check into the `matcher.find()` check, so we don't scan the whole document multiple times unless we find a `@since` by the pattern first?

nizarbenalla replied:
Thanks for the feedback, fixed.

liach commented:
/label add javadoc

nizarbenalla commented:
Updated based on feedback, I rewrote the `extractSinceVersionFromText` method slightly to deal with `@since` values like 6u25 that I noticed without getting too many nested `if` statements.

liach reviewed (approved):
Looks good.

nizarbenalla commented:
Thanks for the review

/integrate
